### DISCUSSION TITLE
`x-slot`'s inside `x-slot` & support multiple `x-slot` arrays without hardcoded the slot's name

### DISCUSSION THREAD

**Discussion by bayukartiko** [laravel/framework#45319] [Ideas]
Hello there!,

I have a case of using bootstrap accordion, the basic bootstrap accordion according the documentation is like this: [https://getbootstrap.com/docs/5.2/components/accordion](https://getbootstrap.com/docs/5.2/components/accordion)

now in my blade at least like this:
```
<x-features.custom.accordion>
    <x-slot:[item]>
        <x-slot:header>
            title 1
        </x-slot:header>
        <x-slot:body>
            body 1
        </x-slot:body>
    </x-slot:[item]>
    <x-slot:[item]>
        <x-slot:header>
            title 2
        </x-slot:header>
        <x-slot:body>
            body 2
        </x-slot:body>
    </x-slot:[item]>
    ...
</x-features.custom.accordion>
```

> ---
> about multiple `x-slot`

Unfortunately, multiple arrays of `x-slot` names doesn't allowed. like this:
```
<x-features.custom.accordion>
    <x-slot:[item]>
        ...
    </x-slot:[item]>
    <x-slot:[item]>
        ...
    </x-slot:[item]>
    ...
</x-features.custom.accordion>
```

You have to manually hardcoded the slot name instead:
```
<x-features.custom.accordion>
    <x-slot:item_1>
        ...
    </x-slot:item_1>
    <x-slot:item_2>
        ...
    </x-slot:item_2>
    ...
</x-features.custom.accordion>
```
or like this answers via: [https://laracasts.com/discuss/channels/general-discussion/blade-dynamic-component-dynamic-named-slots?page=1&replyId=646487](https://laracasts.com/discuss/channels/general-discussion/blade-dynamic-component-dynamic-named-slots?page=1&replyId=646487)

I think it would be a nice way to make the code a bit easier without manually hardcoded the name again if I want to add more items inside the accordion.

> ---
> about `x-slot`'s inside `x-slot`

Unfortunately, `x-slot`'s inside `x-slot` doesn't allowed. this sample code below will not be detected as a slot or as plain HTML instead.
sample like this:
```
<x-features.custom.accordion>
    <x-slot:item_1>
        <x-slot:header>
            <span>title 1<span>
        </x-slot:header>
        <x-slot:body>
            <div>body 1<div>
        </x-slot:body>
        ...
    </x-slot:item_1>
    ...
</x-features.custom.accordion>
```

I think it would be a great approach if `x-slot`'s inside `x-slot` and multiple arrays of `x-slot` names can be PR to laravel 9.x or 10.x maybe, 

I'll appreciate it if this ideas have various suggestions from you all, and it will be a lot of improvement of this idea and the laravel framework of course.

**Comment by munezaclovis** [comment]
Hey, Did you ever find a solution for this?

I am currently trying to build a context-menu component and it would be nice to use named slots instead of creating a new component just for the content inside.
